Update list of changes since last release in the CHANGELOG (#662)
alphagov/accessible-autocomplete: CHANGELOG.md

## Unreleased

### Breaking changes

#### Verify your code does not rely on polyfills we have now removed

> The code of the accessible autocomplete is no longer transpilled and polyfilled to support IE8-10, matching [our previous announcement](https://github.com/alphagov/design-system-team-internal/issues/561).

Reviewer: ```suggestion
Following on from [our previous announcement](https://github.com/alphagov/design-system-team-internal/issues/561), the accessible autocomplete code is no longer transpilled and polyfilled to support IE8-10.
```

Reviewer 2: Think there's one too many l's in transpiled?

Reviewer: 'tis a better pil than a bitter pill

- `hintClasses` for the suggestion hint (`input` element appearing when the text typed by the user matches the start of an option)
- `menuClasses` for the `ul` element listing the options

You can also add custom classes to the `ul` listing the options through the new `menuAttributes` option, providing a `className` or `class` property.

This change was introduced in:
- [pull request #602: Add `inputClasses` option](https://github.com/alphagov/accessible-autocomplete/pull/602). Thanks to @andreyyudin for contributing this change.
- [pull request #649: Update options for styling component inputs](https://github.com/alphagov/accessible-autocomplete/pull/649)
- [pull request #650: Update ways of configuring menu attributes](https://github.com/alphagov/accessible-autocomplete/pull/650)

#### New class on the component's status

Use the new `autocomplete__status` class in your CSS (or `<CSS_NAMESPACE>__status` if you set [the `cssNamespace` option](https://github.com/alphagov/accessible-autocomplete?tab=readme-ov-file#cssnamespace-default-autocomplete)) to customise the styles of the component's status element (hidden element that makes announcements to assistive technologies).

> This change was introduced in [pull request #453: Add className attribute to status component](https://github.com/alphagov/accessible-autocomplete/pull/453) (merged via [pull request #620](https://github.com/alphagov/accessible-autocomplete/pull/620)). Thanks to @lennym for contributing this change.

Author: Felt best to acknowledge the contributor's PR first, to highlight their contribution, and then link to the PR we had to open for merging because we couldn't push on their branch.

Author: > Like GitHub does when you reference something in a new issue.

Oh, do you have an example of this, I'm not sure I understand what you mean?

Reviewer 2: https://github.com/alphagov/govuk-frontend/issues/1845#issue-643089233

### Fixes

- [Pull request #496: Handle query change when the text length does not change](https://github.com/alphagov/accessible-autocomplete/pull/496) (merged through [pull request #621](https://github.com/alphagov/accessible-autocomplete/pull/621)). Thanks to @archferns for contributing this change.
- [Pull request #591: Add `menuAttributes` to fix Axe's 'aria-input-field-name' error](https://github.com/alphagov/accessible-autocomplete/pull/591). Thanks to @mchughbri for contributing this change.
- [Pull request #600: Fix null aria attributes](https://github.com/alphagov/accessible-autocomplete/pull/600). Thanks to @mark-roberts-ho for contributing this change.

> - [Pull request #606: Correct description of tStatusSelectedOption i18n option in README and tweak some wording for consistency with other parts of the README](https://github.com/alphagov/accessible-autocomplete/pull/606). Thanks to @ellamdav for contributing this change.

Author: This is a docs fix, rather than a code fix, but it felt important to acknowledge that kind of contribution as well.

## Unreleased

### Breaking changes

#### Verify your code does not rely on removed polyfills

Following on from [our previous announcement](https://github.com/alphagov/design-system-team-internal/issues/561), the accessible autocomplete code is no longer transpiled and polyfilled to support IE8-10.

However, because the polyfills create or extend global objects, you might have other code in your service unintentionally relying on the inclusion of these polyfills. You might need to introduce your own polyfills or rewrite your JavaScript to avoid using the polyfilled features.

This change was introduced in [pull request #612: Update packages, configs + Node.js 20](https://github.com/alphagov/accessible-autocomplete/pull/612).

> #### Check for minor visual changes in the rendering of the component

Reviewer: is it a particular component? if not, then this possibly needs to say 'a component' or 'the components' or 'any component' depending on what is the most accurate.

Author: Yes, the 'the' refers to the Accessible autocomplete component. It's the only component in this repository, so I think 'the' would be the most accurate in this case.

Reviewer: ok. then you can ignore all my suggestions. I'll see if i can remove/resolve them

> We've made some style adjustments to the stylesheet shipped with the component to make its colours, height, line-height and padding match those of the latest version of GOV.UK Frontend's `<input>`.

Reviewer: ```suggestion
We've made some style adjustments to the stylesheet shipped with components to make colours, height, line-height and padding match those of the latest version of GOV.UK Frontend's `<input>`.
```

If you're using this stylesheet:
- check if our style update affects the rendering of the component in your service, and adjust if necessary
- if you were adding to the default style's to match GOV.UK Frontend's input,
look for CSS declarations you may now be able to remove

This change was introduced in [pull request #644: Align CSS styles with GOV.UK Frontend](https://github.com/alphagov/accessible-autocomplete/pull/644).

### New features

> #### Options to add classes to parts of the component

Reviewer: same query with the mentions of the component in this section. Does it work if it says the following:
```suggestion
#### Options to add classes to parts of components
```

Use these new options to add your own classes to parts of the component:
- `inputClasses` for the `input` element
- `hintClasses` for the suggestion hint (`input` element appearing when the text typed by the user matches the start of an option)
- `menuClasses` for the `ul` element listing the options

You can also add custom classes to the `ul` listing the options through the new `menuAttributes` option, providing a `className` or `class` property.

This change was introduced in:
- [pull request #602: Add `inputClasses` option](https://github.com/alphagov/accessible-autocomplete/pull/602). Thanks to @andreyyudin for contributing this change.
- [pull request #649: Update options for styling component inputs](https://github.com/alphagov/accessible-autocomplete/pull/649)
- [pull request #650: Update ways of configuring menu attributes](https://github.com/alphagov/accessible-autocomplete/pull/650)

> #### New class on the component's status

Reviewer: ```suggestion
#### New class on a component's status
```
